### PR TITLE
Provide list of editable fields via OCS API

### LINKED ISSUE #7983: Provisioning API: expose what can be edited
For https://github.com/nextcloud/android/issues/1925

Basically to allow editing the user info fields from the clients they have to know which fields are there ;)

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #7983

CC: @tobiasKaminsky

### CODE REVIEW

File: apps/provisioning_api/appinfo/routes.php [hunk 1]
```diff
@@ -44,6 +44,7 @@
 		['root' => '/cloud', 'name' => 'Users#addUser', 'url' => '/users', 'verb' => 'POST'],
 		['root' => '/cloud', 'name' => 'Users#getUser', 'url' => '/users/{userId}', 'verb' => 'GET'],
 		['root' => '/cloud', 'name' => 'Users#getCurrentUser', 'url' => '/user', 'verb' => 'GET'],
+		['root' => '/cloud', 'name' => 'Users#getEditableField', 'url' => '/user/fields', 'verb' => 'GET'],
```
Comment (rullzer) [inline apps/provisioning_api/appinfo/routes.php]: Suggestions for a better route are welcome ;)

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 2]
```diff
@@ -292,6 +292,38 @@ protected function getUserData($userId) {
...
+
+		// Editing self (display, email)
+		if ($this->config->getSystemValue('allow_user_to_change_display_name', true) !== false) {
+			$permittedFields[] = AccountManager::PROPERTY_DISPLAYNAME;
+			$permittedFields[] = AccountManager::PROPERTY_EMAIL;
+		}
+
+		$permittedFields[] = 'password';
+		if ($this->config->getSystemValue('force_language', false) === false) {
+			$permittedFields[] = 'language';
```
Comment (rullzer) [inline apps/provisioning_api/lib/Controller/UsersController.php]: should we then also list the available languages?

**Comment by tobiasKaminsky** [comment]
@przybylski this is for you :-)

**Comment by MorrisJobke** [comment]
@rullzer What about tests?

**Comment by rullzer** [comment]
@MorrisJobke will do once we all agree this is the API we want :wink:

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 3]
```diff
@@ -292,6 +292,38 @@ protected function getUserData($userId) {
...
+	public function getEditableField() {
+		$permittedFields = [];
+
+		// Editing self (display, email)
+		if ($this->config->getSystemValue('allow_user_to_change_display_name', true) !== false) {
+			$permittedFields[] = AccountManager::PROPERTY_DISPLAYNAME;
+			$permittedFields[] = AccountManager::PROPERTY_EMAIL;
+		}
+
+		$permittedFields[] = 'password';
```
Comment (juliushaertl) [inline apps/provisioning_api/lib/Controller/UsersController.php]: Changing passwords is not always possible, like on some ldap setups.

**Comment by rullzer** [comment]
Now with tests

**Review by juliusknorr — APPROVED** [review]

**Comment by rullzer** [comment]
Review time ;)

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**